### PR TITLE
Remove e2e-openstack-ccm-install jobs

### PR DESCRIPTION
**Pull request by MaysaMacedo**
As external cloud provider is now the default provider used
for the OCP cluster running on OpenStack, there is no need to
have jobs enabling it as tech preview on master and later releases.
Also, jobs running on previous releases can now be removed.

### CODE REVIEW

**Comment by MaysaMacedo** [comment]
/hold
let's wait for MCO PR to merge first https://github.com/openshift/machine-config-operator/pull/3270

**Comment by pierreprinetti** [comment]
Nice, that will give some more room in the CI.

Just a couple nits:
* please remove changes to `priv`; those are handled by automation anyway
* please remove `e2e-openstack-techpreview-parallel`, `e2e-openstack-techpreview-serial` and `e2e-openstack-ccm-install` from `shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.11.yaml`

File: ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml [hunk 1]
```diff
@@ -122,7 +122,7 @@ tests:
       BASE_DOMAIN: shiftstack.devcluster.openshift.com
       CONFIG_TYPE: minimal
       TEST_SUITE: experimental/reliability/minimal
-    workflow: openshift-e2e-openstack-ccm
+    workflow: openshift-e2e-openstack-ipi
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml:125]: Should we rename the test?

**Review by JoelSpeed — COMMENTED** [review]
Should we be dropping the CCM tests from older releases that we are still supporting?

**Comment by pierreprinetti** [comment]
We can also remove the workflow in ci-operator/step-registry/openshift/e2e/openstack/ccm

**Comment by MaysaMacedo** [comment]
> ci-operator/step-registry/openshift/e2e/openstack/ccm

I tried doing that. It will fail since we have the jobs on openshift-priv. Once this merges and those private jobs are synced, perhaps that can be moved?

File: ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml (hunk 1 above)
Comment (MaysaMacedo) [reply ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml:125]: right, I will leave the default of that workflow to be run, which is conformance/parallel plus openstack specific tests.

**Comment by pierreprinetti** [comment]
> > ci-operator/step-registry/openshift/e2e/openstack/ccm
> 
> I tried doing that. It will fail since we have the jobs on openshift-priv using it. Once this merges and those private jobs are synced, perhaps that can be removed?

Uh-oh right

**Comment by pierreprinetti** [comment]
There is still `e2e-openstack-ccm` in `ci-operator/config/openshift/cluster-cloud-controller-manager-operator`. Is that expected?

Wow we had kind of a coverage!

**Comment by MaysaMacedo** [comment]
> There are still `e2e-openstack-ccm` in `ci-operator/config/openshift/cluster-cloud-controller-manager-operator`. Is that expected?
> 
>     * openshift-cluster-cloud-controller-manager-operator-release-4.13.yaml
> 
>     * openshift-cluster-cloud-controller-manager-operator-release-4.12.yaml
> 
>     * openshift-cluster-cloud-controller-manager-operator-master.yaml
> 
> 
> The tests could be called `e2e-openstack` there now that you've set the default workflow, if I understand Joel's comment.
> 
> Wow we had kind of a coverage!

done.

**Comment by JoelSpeed** [comment]
You've removed CI from the techpreview releases which means if we decide to backport any fixes, we won't be able to test them. I would have expected us to leave the 4.10 and 4.11 branches alone until they are dropped from support. What's the plan?

**Comment by MaysaMacedo** [comment]
> You've removed CI from the techpreview releases which means if we decide to backport any fixes, we won't be able to test them. I would have expected us to leave the 4.10 and 4.11 branches alone until they are dropped from support. What's the plan?

Since CCM is in tech preview for versions prior to 4.12 and consequently not fully supported by Red Hat, we thought it would be ok to remove those jobs to reduce the amount of jobs we run in the CI. Also, I thought the original idea when adding those jobs was to start testing CCM prior to making it GA. Do you think it's better to remain the presumit jobs? Perhaps those can be left as optional ones

**Comment by JoelSpeed** [comment]
My concern is whether we are likely to backport anything to those branches or not. Realistically, there's no harm in leaving the configuration there for presubmits as they will only run if we do decide to backport. If we never create a PR to one of those branches, they will never run

File: ci-operator/config/openshift/cloud-provider-openstack/openshift-cloud-provider-openstack-release-4.10.yaml [hunk 2]
```diff
@@ -95,18 +95,6 @@ tests:
         installed on the cluster shouldn't report any alerts in firing state apart
         from Watchdog and AlertmanagerReceiversNotConfigured
     workflow: openshift-e2e-openstack-ccm
-- as: e2e-openstack-ccm-install
```
Comment (MaysaMacedo) [inline ci-operator/config/openshift/cloud-provider-openstack/openshift-cloud-provider-openstack-release-4.10.yaml:98]: This seemed duplicate of the above job as it also create a featuregate. So, I removed it.

Comment (JoelSpeed) [reply ci-operator/config/openshift/cloud-provider-openstack/openshift-cloud-provider-openstack-release-4.10.yaml:98]: Are the differences not that one bootstraps as techpreview and the other bootstraps and then changes over to tech preview once bootstrapped? So they test slightly different configurations

File: ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml [hunk 3]
```diff
@@ -114,15 +114,14 @@ tests:
...
     workflow: openshift-e2e-gcp-ccm-install
-- as: e2e-openstack-ccm
+- as: e2e-openstack
   optional: true
   steps:
     cluster_profile: openstack-vh-mecha-central
     env:
       BASE_DOMAIN: shiftstack.devcluster.openshift.com
       CONFIG_TYPE: minimal
-      TEST_SUITE: experimental/reliability/minimal
```
Comment (mandre) [inline ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-master.yaml:124]: If we're changing the test suite for the openstack job, we should at least mention it in the commit message. Ideally this should be a separate commit so we can revert if needed.

**Review by mandre — COMMENTED** [review]
Thanks for cleaning up the jobs that are no longer relevant, hopefully this gives a bit of fresh air to our ci infra.

**Review by mandre — COMMENTED** [review]
/lgtm

File: ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-release-4.9.yaml [hunk 4]
```diff
@@ -83,15 +83,6 @@ tests:
   steps:
     cluster_profile: azure4
     workflow: openshift-e2e-azure-ccm-install
-- as: e2e-openstack-ccm
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-release-4.9.yaml:86]: I think we should be keeping this one right?

Comment (mandre) [reply ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-release-4.9.yaml:86]: If you think there's a chance we'll backport changes to CCM in 4.9, then yeah. What's the support status of CCCMO in 4.9?

Comment (MaysaMacedo) [reply ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-release-4.9.yaml:86]: The docs on 4.9 mention support as Tech preview for openstack, perhaps it makes sense?

https://docs.openshift.com/container-platform/4.9/operators/operator-reference.html#cluster-cloud-controller-manager-operator_platform-operators-ref

Comment (MaysaMacedo) [reply ci-operator/config/openshift/cluster-cloud-controller-manager-operator/openshift-cluster-cloud-controller-manager-operator-release-4.9.yaml]: I agree. Done
